website/source/docs/job-specification/spread.html.md

```diff
@@ -0,0 +1,117 @@
+---
+layout: "docs"
+page_title: "spread Stanza - Job Specification"
+sidebar_current: "docs-job-specification-spread"
+description: |-
+  The "spread" stanza allows operators to configure desired target percentages for allocations according to node attributes or metadata.
+  Spread may be specified at the job, group, or task levels for ultimate flexibility.
+  More than one spread stanza may be specified with relative weights between each.
+---
+
+# `spread` Stanza
+
+<table class="table table-bordered table-striped">
+  <tr>
+    <th width="120">Placement</th>
+    <td>
+      <code>job -> **spread**</code>
+      <br>
+      <code>job -> group -> **spread**</code>
+      <br>
+      <code>job -> group -> task -> **spread**</code>
+    </td>
+  </tr>
+</table>
+
+The `spread` stanza allows operators to specify target percentages of allocations according to
```

Comment:
I think we need to rework this. I think adding more examples will be very important. We need to:
* motivate what spread is for
* give some concrete use cases
* Explain what it does when given targets and when it is just given an attribute
* Explain what it does when the set of given targets aren't exhaustive

Reply:
Taking a stab:

The `spread` stanza allows operators to influence the placement of allocations in a task group with a count greater than one. The operator can specify a node attribute such as datacenter, availability zone, or even rack in a physical datacenter to spread the allocations over. By default, when using spread the scheduler will attempt to place allocations equally among the available values of the given target.

An example of this, is imagine we have three datacenters: ["us-east1", "us-east2", and "us-west1"] and a task group of `count = 6`. If we provide a spread stanza as seen below, Nomad will attempt to place 2 allocations in each datacenter.

```
spread {
  attribute = "${node.datacenter}"
  weight    = 100
}
```

Alternatively, if an operator desires a weighted spread the spread stanza can be augmented with targets. The below example would place 3 of the allocations in "us-east1", and then spread the rest among the other two datacenters.

```
spread {
  attribute = "${node.datacenter}"
  weight    = 100

  target "us-east1" {
    percent = 50
  }
}
```

Reply:
@dadgar I kept the docs very basic because we are also writing a  guide - https://github.com/hashicorp/nomad/pull/5001/files. I haven;t had a chance to review that yet, but that's where I'd imagine more content. 

Will add more examples here though, particularly the one around one target specified when there are many more potential values.